src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.PowerFx.Core.Functions.Delegation;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    // Supports delegation information for CDP connectors
+    public abstract class TableDelegationInfo
+    {
+        // Defines unsortable columns or columns only supporting ascending ordering
+        public SortRestrictions SortRestriction { get; init; }
+
+        // Defines columns that cannot be sorted and required properties
+        public FilterRestrictions FilterRestriction { get; init; }
+
+        // Used to indicate whether this table has selectable columns
+        public SelectionRestrictions SelectionRestriction { get; init; }
+
+        // Defines ungroupable columns
+        public GroupRestrictions GroupRestriction { get; init; }
+
+        // Filter functions supported by all columns of the table
+        // Those entries cumulate with the onee defined at column level
+        // Possible strings are defined in DelegationMetadataOperatorConstants
+        // Values not in this list are ignored
+        public IEnumerable<string> FilterFunctions { get; init; }
+
+        // Filter functions supported by the table
+        public IEnumerable<string> FilterSupportedFunctions { get; init; }
+
+        // Defines paging capabilities
+        internal PagingCapabilities PagingCapabilities { get; init; }
+
+        // Defining per column capabilities
+        public IReadOnlyCollection<KeyValuePair<string, ColumnCapabilitiesBase>> ColumnsCapabilities { get; init; }
```

Comment:
As far as public interface is concerned, I think a better approach here is a list of column names and a function to map column name to capability.  Requiring the implementer to materialize this as a whole collection upfront could be a barrier for some use cases

Reply:
We can make this one internal as it's only going to be used in CDP case
DV doesn't need it
ServiceNow will override TableDelegationInfo and override GetColumnCapability to get ColumnCapabilitiesDefinition